docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+
+#### Using the injector to customise a core component
+
+Let's make a more awesome text field. Because the `TextField` component is fetched 
+through the injector, we can override it and augment it with our own functionality.
+
+In this example, we'll add a simple character count below the text field.
+
+First, let's create our higher order component.
+__my-module/js/components/CharacterCounter.js__
+```js
+import React from 'react';
+
+const CharacterCounter = (TextField) => (props) => {
+    return (
+        <div>
+            <TextField {...props} />
+            <small>Character count: {props.value.length}</small>
+        </div>
+    );
+}
+
+export default CharacterCounter;
+```
+
+Now let's add this higher order component to the injector. 
+
+__my-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import CharacterCounter from './components/CharacterCounter';
+
+Injector.transform('my-transformation', (update) => {
+  update('TextField', CharacterCounter);
+});
+```
+
+Much like the configuration layer, we need to specify a name for this transformation. This
+will help other modules negotiate their priority over the injector in relation to yours.
+
+The second parameter of the `transform` argument is a callback which receives a `update()` function
+that allows you to mutate the DI container with a wrapper for the component. Remember, this function does not _replace_
+the component -- it enhances it with new functionality.
+
+The last thing we'll have to do is transpile our code and load the resulting bundle file
+into the admin page.
+
+__my-module/\_config/config.yml__
+
+```yaml
+    ---
+    Name: my-module
+    ---
+    SilverStripe\Admin\LeftAndMain:
+      extra_requirements_javascript:
+        # The name of this file will depend on how you've configured your build process
+        - 'my-module/js/dist/main.bundle.js'
+```
+Now that the customisation is applied, our text fields look like this:
+
+![](../../../_images/react-di-1.png)
+
+Let's add another customisation to TextField. If the text goes beyond a specified
+length, let's throw a warning in the UI.
+
+__my-module/js/components/TextLengthChecker.js__
+```js
+const TextLengthCheker = (TextField) => (props) => {  
+  const {limit, value } = props;
+  const invalid = limit !== undefined && value.length > limit;
+
+  return (
+    <div>
+      <TextField {...props} />
+      {invalid &&
+        <span style={{color: 'red'}}>
+          {`Text is too long! Must be ${limit} characters`}
+        </span>
+      }
+    </div>
+  );
+}
+
+export default TextLengthChecker;
+```
+
+We'll apply this one to the injector as well, but let's do it under a different name.
+For the purposes of demonstration, let's imagine this customisation comes from another
+module.
+
+__my-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import TextLengthChecker from './components/TextLengthChecker';
+
+Injector.transform('my-other-transformation', (update) => {
```

Comment:
As best I can see you've deleted the CharacterCounter reference in this example. Does it need to be added back?